extension/src/helpers/muxedAddress.ts

```diff
@@ -0,0 +1,205 @@
+    // Check if 'to_muxed' parameter exists in properties (CAP-0067 pattern)
+    // Use 'in' operator to check if key exists, regardless of value
+    if (argsProperties && "to_muxed" in argsProperties) {
+      return true;
+    }
+
+    // Also check if to_muxed is in the required array (even if property doesn't exist)
+    if (Array.isArray(required) && required.includes("to_muxed")) {
+      return true;
+    }
+
+    return false;
+  } catch (error) {
+    // If we can't fetch the spec, assume no muxed support for safety
+    return false;
+  }
+}
+
+export interface MemoDisabledStateParams {
+  targetAddress: string;
+  contractId?: string;
+  networkDetails?: NetworkDetails;
+  t: (key: string) => string;
+}
+
+export interface MemoDisabledState {
+  isMemoDisabled: boolean;
+  memoDisabledMessage?: string;
+}
+
+/**
+ * Determines if memo should be disabled for a transaction.
+ * Disables memo for all M addresses (memo is encoded in the address).
+ *
+ * @param params Parameters for determining memo disabled state
+ * @returns Object indicating if memo is disabled and optional message
+ */
+export function getMemoDisabledState(
+  params: MemoDisabledStateParams,
+): MemoDisabledState {
+  const { targetAddress, contractId, networkDetails, t } = params;
+
+  // Disable memo for all M addresses (memo is encoded in the address)
+  if (isMuxedAccount(targetAddress)) {
+    return {
+      isMemoDisabled: true,
+      memoDisabledMessage: t("Memo is disabled for this transaction"),
+    };
+  }
+
+  if (!contractId || !networkDetails) {
+    return { isMemoDisabled: false, memoDisabledMessage: undefined };
+  }
+
+  if (isContractId(targetAddress)) {
+    return {
+      isMemoDisabled: true,
+      memoDisabledMessage: t("Memo is not supported for this operation"),
+    };
+  }
+
+  if (!isValidStellarAddress(targetAddress) && !isMuxedAccount(targetAddress)) {
+    return {
+      isMemoDisabled: true,
+      memoDisabledMessage: t("Memo is not supported for this operation"),
+    };
+  }
+
+  // For Soroban transactions (custom tokens), memo is supported for G addresses
+  return { isMemoDisabled: false, memoDisabledMessage: undefined };
+}
+
+export interface DetermineMuxedDestinationParams {
+  recipientAddress: string;
+  transactionMemo?: string;
+  contractSupportsMuxed: boolean;
+}
+
+/**
+ * Determines the final destination address for a transaction.
+ * For contracts that support muxed addresses (CAP-0067), creates a muxed address
+ * from a G address + memo if needed.
+ *
+ * @param params Parameters for determining muxed destination
+ * @returns The final destination address (may be muxed if memo was provided and contract supports it)
+ */
+export function determineMuxedDestination(
+  params: DetermineMuxedDestinationParams,
+): string {
+  const { recipientAddress, transactionMemo, contractSupportsMuxed } = params;
+
+  const isRecipientGAddress =
+    isValidStellarAddress(recipientAddress) && !isContractId(recipientAddress);
```

Comment:
`isValidStellarAddress` will actually return true for a M address or a federated address - so this actually isn't necessarily a G address